deployment/confStorage/external_version_control/git_storage.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+import git
+import sys
+import time
+import logging
+
+
+from ...paiLibrary.common import file_handler
+from ...paiLibrary.common import directory_handler
+
+
+
+class git_storage:
+
+    def __init__(self, storage_configuration, local_store = "/tmp/pai-conf-sync/git-storage"):
+
+        self.repo_url = storage_configuration["url"]
+        self.branch = storage_configuration["branch"]
+        self.path = storage_configuration["path"]
+
+        self.time = str(int(time.time()))
+        self.logger = logging.getLogger(__name__)
+        self.local_store = "{0}-{1}".format(local_store, self.time)
```

Comment:
why not using [`tempfile`](https://docs.python.org/2/library/tempfile.html)

Reply:
We will have to remove the temp directory created by ```tempfile.mkdtemp([suffix=''[, prefix='tmp'[, dir=None]]]) ```.  And I think it's same as what we have done.